### PR TITLE
LPS-149503: Expose ExternalReferenceCode in Structured Content using AssetLibraryId

### PR DESCRIPTION
**Pull request by sergiojimcos**
**What is new?**
- Expose **External Reference Code** in `Structured Content`.
- Endpoint implementation of `GET`, `UPSERT` and `DELETE` by **External Reference Code** using **AssetLibraryId**.
- Including some tests in API part.

## API
In the API section, this PR contains the new endpoints to operate using the ERC:

Screenshot 2022-05-31 at 17 50 13

To see this new endpoint you have to follow these steps:

1. Deploy `journal-api` and `journal-service` in this order.
2. Deploy `headless-delivery-api` and `headless-delivery-impl` in this order.
3. Go to `/o/api`
4. Go to `StructuredContent`

**JIRA Ticket:**
https://issues.liferay.com/browse/LPS-149503

### CODE REVIEW

File: modules/apps/headless/headless-delivery/headless-delivery-impl/src/main/java/com/liferay/headless/delivery/internal/resource/v1_0/StructuredContentResourceImpl.java [hunk 1]
```diff
@@ -263,9 +288,11 @@ public StructuredContent getSiteStructuredContentByExternalReferenceCode(
 			Long siteId, String externalReferenceCode)
 		throws Exception {
 
-		return _getStructuredContent(
+		JournalArticle journalArticle =
```
Comment (LuismiBarcos) [inline modules/apps/headless/headless-delivery/headless-delivery-impl/src/main/java/com/liferay/headless/delivery/internal/resource/v1_0/StructuredContentResourceImpl.java:291]: Why this refactor?? 🤔

**Review by LuismiBarcos — CHANGES_REQUESTED** [review]
Really really good! 👏🏻👏🏻👏🏻  Just review a small comment because I don't a little refactor.
Again, congrats! A really good job 🙂

**Comment by sergiojimcos** [comment]
ci:test:sf

**Comment by sergiojimcos** [comment]
ci:test:relevant

**Comment by sergiojimcos** [comment]
ci:test:relevant

File: modules/util/portal-tools-rest-builder/src/main/resources/com/liferay/portal/tools/rest/builder/dependencies/base_resource_test_case.ftl [hunk 2]
```diff
@@ -1710,6 +1710,9 @@ public abstract class Base${schemaName}ResourceTestCase {
 
 															<#if stringUtil.equals(javaMethodParameter.parameterName, "siteId")>
 																put("siteKey", <@getQuotedString unquotedString="testGraphQL${javaMethodSignature.methodName?cap_first}_get${javaMethodParameter.parameterName?cap_first}()" />);
+															<#elseif stringUtil.equals(javaMethodParameter.parameterName, "assetLibraryId")>
+																put("${javaMethodParameter.parameterName}",
+																	<@getQuotedString unquotedString="testGraphQL${javaMethodSignature.methodName?cap_first}_get${javaMethodParameter.parameterName?cap_first}()" />);
```
Comment (ccorreagg) [inline modules/util/portal-tools-rest-builder/src/main/resources/com/liferay/portal/tools/rest/builder/dependencies/base_resource_test_case.ftl:1715]: Maybe better in the same line to maintain the same format

**Comment by sergiojimcos** [comment]
ci:test:sf

**Comment by sergiojimcos** [comment]
ci:test:relevant

**Review by ccorreagg — APPROVED** [review]
Great job!! I hope you enjoyed your first contact with REST Builder :smile:

**Review by LuismiBarcos — APPROVED** [review]
LGTM :)

**Comment by LuismiBarcos** [comment]
ci:forward

**Pull request closed without merging**